### PR TITLE
Select tenant

### PR DESCRIPTION
**Pull request by aviadl**
+tests
related to https://github.com/descope/etc/issues/3078 Added an API call for selecting a tenant and placing it on the jwt as a custom claim

### CODE REVIEW

File: descope/sdk/auth.go [hunk 1]
```diff
@@ -313,6 +313,12 @@ type Authentication interface {
 	// the specified roles for a specific tenant.
 	ValidateTenantRoles(token *descope.Token, tenant string, roles []string) bool
 
+	// SelectTenantWithRequest - Use to add to the jwt a dedicated claim to mark on which tenant this user authenticated now
```
Comment (asafshen) [inline descope/sdk/auth.go:316]: comment is great, here is an improvement suggestion though

```suggestion
	// SelectTenantWithRequest - Adds a dedicated claim to the JWTs to indicate the tenant on which the user is currently authenticated
```

File: descope/sdk/auth.go [hunk 2]
```diff
@@ -313,6 +313,12 @@ type Authentication interface {
...
+	SelectTenantWithRequest(tenantID string, request *http.Request, w http.ResponseWriter) (*descope.AuthenticationInfo, error)
+
+	// SelectTenantWithToken - Use to add to the jwt a dedicated claim to mark on which tenant this user authenticated now
```
Comment (asafshen) [inline descope/sdk/auth.go:319]: with respect to the above

```suggestion
	// SelectTenantWithToken - Adds a dedicated claim to the JWTs to indicate the tenant on which the user is currently authenticated
```

**Review by asafshen — APPROVED** [review]
👍

File: descope/sdk/auth.go (hunk 1 above)
Comment (aviadl) [reply descope/sdk/auth.go]: done

File: descope/sdk/auth.go (hunk 2 above)
Comment (aviadl) [reply descope/sdk/auth.go]: done

**Review by asafshen — APPROVED** [review]

**Pull request merged**